src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -143,50 +114,174 @@ public class RabbitMQMessageBus : MessageBusBase, IMessageBus {
+            // However, we plugin is not installed this will throw an exception. In that case
+            // we attempt to create regular exchange. If regular exchange also throws and exception 
+            // then trouble shoot the problem.
+            if (!CreateExchange(_publisherChannel)) {
+                // if the initial exchange creation was not successful then we must close the previous connection
+                // and establish the new client connection and model otherwise you will keep recieving failure in creation
+                // of the regular exchange too.
+                _publisherClient = _factory.CreateConnection();
+                _publisherChannel = _publisherClient.CreateModel();
+                CreateExchange(_publisherChannel);
+            }
+            _logger.Trace("The unique channel number for the publisher is : {channelNumber}", _publisherChannel.ChannelNumber);
+        }
 
-            if (_subscriberClient.IsOpen)
-                _subscriberClient.Close();
-            _subscriberClient.Dispose();
+        /// <summary>
+        /// Initialize the subscriber
+        /// </summary>
+        private void InitSubscriber() {
+            _subscriberClient = CreateConnection();
+            _subscriberChannel = _subscriberClient.CreateModel();
+            // If InitPublisher is called first, then we will never come in this if clause.
+            if (!CreateExchange(_subscriberChannel)) {
+                _subscriberClient = _factory.CreateConnection();
+                _subscriberChannel = _subscriberClient.CreateModel();
+                CreateExchange(_subscriberChannel);
+            }
+            _logger.Trace("The unique channel number for the subscriber is : {channelNumber}", _subscriberChannel.ChannelNumber);
+        }
+
+        private bool CreateExchange(IModel model) {
+            bool success = true;
+            try {
+                if (_delayedExchangePluginEnabled) {
+                    //This exchange is a delayed exchange (direct).You need rabbitmq_delayed_message_exchange plugin to RabbitMQ
+                    // Disclaimer : https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/ . Please read the *Performance
+                    // Impact* of the delayed exchange type.
+                    var args = new Dictionary<string, object> { { "x-delayed-type", ExchangeType.Fanout } };
+                    model.ExchangeDeclare(_exchangeName, "x-delayed-message", true, false, args);
+                }
+                else {
+                    // If you don't need to delay messages, then use the actual exchange
+                    model.ExchangeDeclare(_exchangeName, ExchangeType.Fanout, true, false, null);
+                }
+            }
+            catch (OperationInterruptedException o) {
+                if (o.ShutdownReason.ReplyCode == 503) {
+                    _delayedExchangePluginEnabled = false;
+                    success = false;
```

Comment:
I actually split create exchange method as CreateDelayedExchange and CreateRegularExchange